core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,281 @@
+ * manual polling, wait for polling to complete and get status of current polling.
+ * <p>
+ * Since auto polling is turned <b>on</b> by default. If some scenario requires to disable this feature.
+ * It can be done by calling setAutoPollingEnabled(false) function.
+ *
+ * <p><strong>Implementation of Long Running Operations</strong></p>
+ *
+ * @param <T>
+ */
+
+public class Poller<T> {
+
+    private String nullValueNotAllowedFormat = "Null value for %s not allowed.";
+
+    /*
+     * pollOperation is a Function that takes the previous PollResponse, and
+     * returns a new PollResponse to represent the current state
+     */
+    private Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation;
+
+    /*
+     * Various configuration options to create poller object.
+     */
+    private PollerOptions pollerOptions;
+
+    /*
+     * This will save last poll response.
+     */
+    private PollResponse<T> pollResponse;
+
+    /*
+     * This will be called when cancel operation is triggered.
+     */
+    private Consumer<Poller> cancelOperation;
+
+    /*
+     * Indicate to poll automatically or not when poller is created.
+     * default value is false;
+     */
+    private boolean autoPollingEnabled;
+
+    /*
+     * Indicate if cancel is initiated.
+     */
+    private boolean cancelInitiated;
+
+    private final Flux<PollResponse<T>> fluxHandle;
+
+    /*
+     * Since constructor create a subscriber and start auto polling.
+     * This handle will be used to dispose the subscriber when
+     * client disable auto polling.
+     */
+    private Disposable fluxDisposable;
+
+    /**
+     * Create a Poller object. The polling starts immediately by default.
+     * The poll interval would defined by retryAfter value in {@link PollResponse}.
+     * In absence of retryAfter, the poller will use pollInterval defined in {@link PollerOptions}.
+     *
+     * @param pollerOptions Not null configuration options for poller.
+     * @param pollOperation to be called by poller. It should never return {@code null}. The response should always have valid {@link com.azure.core.polling.PollResponse.OperationStatus}
+     * @see PollerOptions
+     * @see PollResponse
+     */
+    public Poller(PollerOptions pollerOptions, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation) {
+
+        validateAndThrow(pollerOptions, "pollerOptions");
+        validateAndThrow(pollOperation, "pollOperation");
+
+        this.pollerOptions = pollerOptions;
+        this.pollOperation = pollOperation;
+        pollResponse = new PollResponse<>(PollResponse.OperationStatus.NOT_STARTED, null);
+
+        fluxHandle = sendPollRequestWithDelay()
+            .flux()
+            .repeat()
+            .takeUntil(pollResponse -> pollResponse.isDone())
+            .share();
+
+        // auto polling start here
+        setAutoPollingEnabled(true);
```

Comment:
It is dangerous calling a public method here that is non-final - subclasses of Poller may override this method to not perform the expected functionality. Consider making Poller final or taking an alternate approach to ensure the correct behavior is implemented.

Reply:
I will start starting of subscriber in constructor instead of calling setAutoPollingEnabled